### PR TITLE
Enforcer rule to work with "bundle" packaging

### PR DESCRIPTION
**Pull request by suztomo**
Towards #781 

Jaxen uses [packaging: bundle.](https://github.com/jaxen-xpath/jaxen/blob/master/core/pom.xml#L15) The previous PR skipped such packaging.

### CODE REVIEW

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java [hunk 1]
```diff
@@ -117,8 +117,8 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
       }
 
       String projectType = project.getArtifact().getType();
-      if (dependencySection == DependencySection.DEPENDENCIES && !"jar".equals(projectType)) {
-        // When checking non-BOM project, not interested in non JAR artifact
+      if (dependencySection == DependencySection.DEPENDENCIES && "pom".equals(projectType)) {
```
Comment (elharo) [inline enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: what about all the other types: ear, war, javadocs, etc.?

https://maven.apache.org/ref/3.6.1/maven-core/artifact-handlers.html

Comment (suztomo) [reply enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: Added exclusion for when not checking non BOM project: pom, java-source, javadoc.

File: enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java [hunk 2]
```diff
@@ -319,4 +337,32 @@ public void testExecute_shouldSkipNonBomPom() throws EnforcerRuleException {
...
+      setupMockDependencyResolution("com.google.appengine:appengine-api-1.0-sdk:1.9.64");
+
+      when(mockProject.getArtifact())
+          .thenReturn(
+              new org.apache.maven.artifact.DefaultArtifact(
+                  "com.google.cloud",
+                  "linkage-checker-rule-test",
+                  "0.0.1",
+                  "compile",
+                  "bundle", // Maven Bundle Plugin uses "bundle" packaging.
```
Comment (suztomo) [inline enforcer-rules/src/test/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRuleTest.java]: Added test

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java [hunk 3]
```diff
@@ -117,11 +127,19 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
...
-        // When checking non-BOM project, not interested in non JAR artifact
-        return;
+      if (readingDependencyManagementSection) {
+        if (!"pom".equals(projectType)) {
+          logger.warn("A BOM should have packaging pom");
+          return;
+        }
+      } else {
+        if (UNSUPPORTED_NONBOM_PACKAGING.contains(projectType)) {
+          // Unless checking BOM project, not interested in pom artifact
```
Comment (elharo) [inline enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: comment still says this is just about poms

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java [hunk 4]
```diff
@@ -64,6 +65,15 @@
 /** Linkage Checker Maven Enforcer Rule. */
 public class LinkageCheckerRule extends AbstractNonCacheableEnforcerRule {
 
+  /**
+   * Maven packaging values known to be irrelevant to Linkage Check for non-BOM project.
+   *
+   * @see <a href="https://maven.apache.org/ref/3.6.1/maven-core/artifact-handlers.html" >Maven
```
Comment (elharo) [inline enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: nit: space before >

Comment (suztomo) [reply enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: Fixed

File: enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java (hunk 3 above)
Comment (suztomo) [reply enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java]: Removed comment. The variable name says enough.

**Pull request merged**